### ISSUE TITLE
Cannot install

### ISSUE THREAD

**Issue by nily-dti** [pantor/frankx#1]
Hi

I'm trying to install your library but I'm stuck in a make error:

```
RUN cd frankx && mkdir build && cd build     && cmake -DREFLEXXES_TYPE=ReflexxesTypeII -DReflexxes_INCLUDE_DIR=RMLTypeII/include/RMLTypeII/ -DReflexxes_LIB_DIR=RMLTypeII/build ..     && make -j8
 ---> Running in 9568e9979c92
-- The CXX compiler identification is GNU 7.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
CMake Error at cmake/FindReflexxes.cmake:10 (target_include_directories):
  Cannot specify include directories for imported target
  "Reflexxes::Reflexxes".
Call Stack (most recent call first):
  CMakeLists.txt:19 (find_package)


CMake Error at cmake/FindReflexxes.cmake:11 (target_link_libraries):
  Cannot specify link libraries for target "Reflexxes::Reflexxes" which is
  not built by this project.
Call Stack (most recent call first):
  CMakeLists.txt:19 (find_package)


-- Configuring incomplete, errors occurred!
See also "/code/frankx/build/CMakeFiles/CMakeOutput.log".
See also "/code/frankx/build/CMakeFiles/CMakeError.log".
The command '/bin/sh -c cd frankx && mkdir build && cd build     && cmake -DREFLEXXES_TYPE=ReflexxesTypeII -DReflexxes_INCLUDE_DIR=RMLTypeII/include/RMLTypeII/ -DReflexxes_LIB_DIR=RMLTypeII/build ..     && make -j8' returned a non-zero code: 1
```

The Docker file to reproduce this is here: https://github.com/dti-research/frankx/blob/master/docker/Dockerfile

I've tried to change the [FindReflexxes.cmake](https://github.com/dti-research/frankx/blob/master/cmake/FindReflexxes.cmake) file to avoid this error, but my changes just gives an error at compile time instead:

```
Step 10/10 : RUN cd frankx && mkdir build && cd build     && cmake -DREFLEXXES_TYPE=ReflexxesTypeII -DReflexxes_INCLUDE_DIR=../RMLTypeII/include/RMLTypeII/ -DReflexxes_LIB_DIR=../RMLTypeII/build -DBUILD_TESTS=OFF ..     && make -j8
 ---> Running in 6f9b8a815891
-- The CXX compiler identification is GNU 7.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Found PythonInterp: /usr/bin/python3.6 (found version "3.6.9") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.6m.so
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- LTO enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /code/frankx/build
Scanning dependencies of target libfrankx
[  8%] Building CXX object CMakeFiles/libfrankx.dir/src/gripper.cpp.o
[ 16%] Building CXX object CMakeFiles/libfrankx.dir/src/robot.cpp.o
[ 25%] Building CXX object CMakeFiles/libfrankx.dir/src/motion_joint.cpp.o
[ 33%] Building CXX object CMakeFiles/libfrankx.dir/src/affine.cpp.o
[ 41%] Building CXX object CMakeFiles/libfrankx.dir/src/reaction.cpp.o
[ 50%] Linking CXX shared library liblibfrankx.so
/usr/bin/ld: cannot find -lReflexxes-NOTFOUND
collect2: error: ld returned 1 exit status
CMakeFiles/libfrankx.dir/build.make:200: recipe for target 'liblibfrankx.so' failed
make[2]: *** [liblibfrankx.so] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/libfrankx.dir/all' failed
make[1]: *** [CMakeFiles/libfrankx.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
The command '/bin/sh -c cd frankx && mkdir build && cd build     && cmake -DREFLEXXES_TYPE=ReflexxesTypeII -DReflexxes_INCLUDE_DIR=../RMLTypeII/include/RMLTypeII/ -DReflexxes_LIB_DIR=../RMLTypeII/build -DBUILD_TESTS=OFF ..     && make -j8' returned a non-zero code: 2
```

**Comment by museum-future** [comment]
This looks like  bad makefile, I need this as well and hack a solution.

**Comment by museum-future** [comment]
You need to make this in the build tree and it compiles without problems.

**Comment by pantor** [comment]
@nily-dti Sorry for the late response! It looks that you are not installing the Reflexxes dependency in your Docker container. You can have a look at our CI pipeline [here](https://github.com/pantor/frankx/blob/master/.github/workflows/ci.yml), you should be able to transfer that easily.

Otherwise, installation can really be improved by using Cmake ExternalProject or similar tools.

**Comment by jakub-kaminski** [comment]
Hi, I have the same problem. Readme instructions do not work for me. I tried to run CMake for Frankx with adjusted instructions from the CI pipeline:

`cmake -DREFLEXXES_TYPE=ReflexxesTypeII -DReflexxes_INCLUDE_DIR=/home/jakub/Documents/RMLTypeII/include/RMLTypeII/ -DReflexxes_LIB_DIR=/home/jakub/Documents/RMLTypeII/build -DFranka_DIR:PATH=/home/jakub/Documents/libfranka/build ..`

Please note that I had to attach information about Libfranka since otherwise it was not found.

My CMake errors are:
```
CMake Error at cmake/FindReflexxes.cmake:10 (target_include_directories):
  Cannot specify include directories for imported target
  "Reflexxes::Reflexxes".

Call Stack (most recent call first):
  CMakeLists.txt:19 (find_package)
```
and
```
CMake Error at cmake/FindReflexxes.cmake:11 (target_link_libraries):
  Cannot specify link libraries for target "Reflexxes::Reflexxes" which is
  not built by this project.
Call Stack (most recent call first):
  CMakeLists.txt:19 (find_package)
```
Everything is installed following the CI pipeline. I have Eigen 3.37, pybind 2.4.3, I have libfranka
Reflexxes library is located here: `/usr/local/lib/libReflexxesTypeII.so` and all the dependencies. Cmake and make and sudo make install were successful for frankx dependencies, but not frankx.

What should I do, where the potential error can be? Can you update frankx installation instructions in the Readme? They are different than CI pipeline, and both cases did not work for me.
Looking forward your reply.
Thank you fo your time and effort.
Regards, Jakub

**Comment by megasus** [comment]
@pantor Any idea if the Reflexxes Type IV library is still available? It seems like it's not recommended to use this library with type II.
Thanks for your effort!

**Comment by pantor** [comment]
@megasus Reflexxes Type IV should be aviable for academic purposes at no cost. You can give Type II a try first, however if discontinuity errors occur you probably need to switch to Type IV.

**Comment by megasus** [comment]
@pantor Thanks for your the quick response. Yes, according to the Reflexxes project page, Type 4 should be available for academic purposes. Unfortunately, it seems like they killed the relevant product/download page: http://www.reflexxes.ws/products/academic-version. Do you happen to know how to get access to this version?

**Comment by fbottarel** [comment]
@pantor I am trying to compile the library as well. Seems like I am having the same issues pointed out above. I tried both instructions listed in the README and those in the CI pipeline, none seem to work. Apparently 
```
CMake Error at cmake/FindReflexxes.cmake:10 (target_include_directories):
  Cannot specify include directories for imported target
  "Reflexxes::Reflexxes".
```
is the source of these issues. Any idea on how to address this?

**Comment by fbottarel** [comment]
Answering my own question here: @megasus @pantor @jakub-kaminski @nily-dti 

Apparently the issue is due to the CMake version being used. [This issue](https://gitlab.kitware.com/cmake/cmake/-/issues/15689) was pointing to a very similar problem, and apparently it has been addressed in recent versions of CMake. I believe all of us were using the default Ubuntu xenial or bionic CMake. 

I made a [Dockerfile](https://github.com/fbottarel/dockerfiles/blob/master/frankx/Dockerfile) that follows all the steps needed to compile and test frankx with Reflexxes type II. It uses the latest version of ubuntu (i.e. Focal). Everything works on my end, feel free to give it a spin.
